test/boost/comparable_bytes_test.cc

```diff
@@ -174,3 +174,33 @@ BOOST_AUTO_TEST_CASE(test_timestamp) {
         return data_value(db_clock::time_point(db_clock::duration(milliseconds)));
     }));
 }
+
+template <std::floating_point fp_type>
+struct floating_point_test_data_generator : test_data_generator {
+    floating_point_test_data_generator() {
+        for (fp_type n : {-1e30f, -1e3f, -1.0f, -0.001f, -1e-30f, -0.0f, 0.0f, 1e-30f, 0.001f, 1.0f, 1e3f, 1e30f,
+                    -std::numeric_limits<float>::min(), std::numeric_limits<float>::min(),
+                    -std::numeric_limits<float>::max(), std::numeric_limits<float>::max(),
+                    -std::numeric_limits<float>::infinity(), std::numeric_limits<float>::infinity(),
+                    std::numeric_limits<float>::quiet_NaN()}) {
+            _test_data.emplace_back(n);
+        }
+
+        // double has a few more test items
+        if constexpr (std::is_same_v<fp_type, double>) {
+            for (fp_type n : std::vector<double>{-1e200, -1e100, 1e100, 1e200,
+                        -std::numeric_limits<double>::min(), std::numeric_limits<double>::min(),
+                        -std::numeric_limits<double>::max(), std::numeric_limits<double>::max()}) {
+                _test_data.emplace_back(n);
+            }
+        }
+    }
+};
+
+BOOST_AUTO_TEST_CASE(test_float) {
+    byte_comparable_test(floating_point_test_data_generator<float>());
+}
+
+BOOST_AUTO_TEST_CASE(test_double) {
+    byte_comparable_test(floating_point_test_data_generator<double>());
```

Comment:
How do we test that the results are compatible with Cassandra?

Reply:
There are currently no tests to verify that. We can probably add them when we implement the trie indexes that uses byte comparable format.

Reply:
This is the only thing remaining now. 

There does not seem to be simple way to do this. I was trying to hack the Cassandra code to write out comparable bytes as values of a table coulmn but the latest version which has the support for byte comparable types writes sstable in nb format, which we do not support yet. I am trying another approach now where I'm generating values and directly writing into ScyllaDB to generate the sstables. I'll update as soon as I'm done.